app/models/operatingsystem.rb

```diff
@@ -190,27 +171,49 @@ def available_loaders
     ["None", "PXELinux BIOS"]
   end
 
-  # sets the prefix for the tfp files based on the os / arch combination
-  def pxe_prefix(arch)
-    "boot/#{self}-#{arch}".tr(" ", "-")
+  # Compatibility method, don't want to break all templates that use it.
+  def medium_uri(host)
+    Foreman::Deprecation.deprecation_warning("1.20", "medium_uri is now accessible through @medium_provider.medium_uri in templates")
+    @medium_provider = Foreman::Plugin.medium_providers.find_provider(host)
+    @medium_provider.medium_uri
+  end
+
+  # sets the prefix for the tfp files based on medium unique identifier
+  def pxe_prefix(medium_provider)
```

Comment:
Can you actually drop the pxe_prefix-based approach and replace it with something more flexible? This pins us with prefix-based naming scheme with one unique ID. We could allow providers to have arbitrary naming scheme pattern. For example CV name followed by ID or something more readable than hash.

Even default provider could include Media name, this is much more readable: `CentOS-local-mirror-JKLfdlkjdfs78_33flkjADSDADAS-linux` what you think?

Reply:
I think `pxe_prefix` is not a very informative name. It returns a path from TFTP root to the filename (with leftmost part of the filename). For example for redhat family, it returns `boot/JKLfdlkjdfs78_33flkjADSDADAS` which later will be appended by `-linux` or `-initrd`.

With this PR we can actually change the filename prefix to whatever we want by changing/overloading `unique_id` function from `MediumProviders::Provider` (base class). It has the whole host visible, so it can construct a name just like it wants. It is the extension point I intend to use in Katello to generate a name based on content view id or whatever unique combination there is.

Reply:
Let me put it this way: can you change the naming conventions to include media name?

Pattern: `Media_name-Hash-Filename`

Example: `CentOS_mirror-JKLfdlkjdfs78-linux`